api/server/sdk/cluster_domains.go

```diff
@@ -0,0 +1,131 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/cluster"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ClusterDomainsServer is an implementation of the gRPC OpenStorageClusterDomains interface
+type ClusterDomainsServer struct {
+	server serverAccessor
+}
+
+func (m *ClusterDomainsServer) cluster() cluster.Cluster {
+	return m.server.cluster()
+}
+
+func (m *ClusterDomainsServer) Enumerate(
+	ctx context.Context,
+	req *api.SdkClusterDomainsEnumerateRequest,
+) (*api.SdkClusterDomainsEnumerateResponse, error) {
+	if m.cluster() == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
```

Comment:
No need to check for admin here?

Reply:
Enumerate and Inspect should be open, so that anyone can view the cluster domains.
Only the activate and deactivate APIs should be controlled.